### PR TITLE
Votes counter cache

### LINKED ISSUE #10: Votes table and vote counter cache
`votes`

`wanted_id` foreign key to the `wanteds` table
`user_id` foreign key to the `users` table

on the `wanteds` table add

`vote_count` - `int`

there should be a db trigger for each insert on the `votes` table that recalculates the related `wanted.votes` tally and sets that counter cace.

No blockers

### PR DESCRIPTION
**Pull request by sylvesterroos**
Closes #10

### CODE REVIEW

**Comment by ex-nix** [comment]
After thinking about it, I think using triggers is not the correct approach here because I don't see the problem with running a simple count(*), while building triggers takes a lot of SQL code and whoever manages the database must be careful to not disable the trigger. Given that the votes table will not be excessively large, I think that we should simply use a join on the votes table whenever we want to know how many votes a wanted has.

**Comment by oliver-kriska** [comment]
You can do materialized view and make code which will update it in Elixir. Views can be connected with Ecto like regular schema so it's nothing special.

**Comment by ex-nix** [comment]
Materialized views are often used in big reporting databases. Here we can do a simple join, so why not do just that? Follow KISS

**Comment by oliver-kriska** [comment]
I know, just suggesting better solution than db triggers how you mention. 
> Materialized views are often used in big reporting databases.
Not really big reporting databases, mostly when it makes sense, it depends on data and queries. You can have small data but too complex query with calculation.

File: mix.lock [hunk 1]
```diff
@@ -12,7 +12,7 @@
...
   "floki": {:hex, :floki, "0.36.2", "a7da0193538c93f937714a6704369711998a51a6164a222d710ebd54020aa7a3", [:mix], [], "hexpm", "a8766c0bc92f074e5cb36c4f9961982eda84c5d2b8e979ca67f5c268ec8ed580"},
   "gettext": {:hex, :gettext, "0.24.0", "6f4d90ac5f3111673cbefc4ebee96fe5f37a114861ab8c7b7d5b30a1108ce6d8", [:mix], [{:expo, "~> 0.5.1", [hex: :expo, repo: "hexpm", optional: false]}], "hexpm", "bdf75cdfcbe9e4622dd18e034b227d77dd17f0f133853a1c73b97b3d6c770e8b"},
-  "heroicons": {:git, "https://github.com/tailwindlabs/heroicons.git", "88ab3a0d790e6a47404cba02800a6b25d2afae50", [tag: "v2.1.1", sparse: "optimized"]},
+  "heroicons": {:git, "https://github.com/tailwindlabs/heroicons.git", "88ab3a0d790e6a47404cba02800a6b25d2afae50", [tag: "v2.1.1", sparse: "optimized", depth: 1]},
```
Comment (bcardarella) [inline mix.lock:15]: can you revert this file change

File: test/elixir_most_wanted/wanteds/wanted.ex [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        Repo.insert!(
+          %Wanted{
+            slug_id: "ecto",
+            user_id: user.id
+          },
+          returning: true
+        )
+
+      vote1 = Repo.insert!(%Vote{wanted_id: wanted.id, user_id: user.id})
+      _vote2 = Repo.insert!(%Vote{wanted_id: wanted.id, user_id: user2.id})
```
[the remark this answers is not in the document]
Comment (ex-nix) [reply test/elixir_most_wanted/wanteds/wanted.ex:68]: just wanted to stay consistent. Removed.

**Comment by ex-nix** [comment]
@bcardarella how do you feel about this though? 

> After thinking about it, I think using triggers is not the correct approach here because I don't see the problem with running a simple count(*), while building triggers takes a lot of SQL code and whoever manages the database must be careful to not disable the trigger. Given that the votes table will not be excessively large, I think that we should simply use a join on the votes table whenever we want to know how many votes a wanted has.

**Comment by bcardarella** [comment]
you want to run `count` on every single read?

**Comment by ex-nix** [comment]
Oops, clicked review request without seeing you already replied.

Yes, that's what I was thinking. To me triggers seem like unneeded complexity but I'd like to hear your thoughts on it :slightly_smiling_face:

**Comment by bcardarella** [comment]
I don't agree

**Comment by bcardarella** [comment]
Sorry was on mobile, let me provide more context.

Most app devs tend to shy away from db layer stuff. We app devs do lean heavily into abstractions and APIs whereas and see the SQL itself as verbose and less maintainable. I'd be willing to wager that the trigger code will sit unchanged and undisturbed for quite some time.

Databases are heavily under utilized when it comes to so many tasks that we end up using the application layer for, and usually app layer is far less performant. Embrace the database :)

Will merge, thank you!

**Pull request merged**